### DISCUSSION TITLE
masks question

### DISCUSSION THREAD

**Discussion by dialuser** [opengeos/segment-geospatial#186] [Q&A] [answered]
This is a question related to hq_sam.py. When I turned on batch=True in SamGeo.generate(...), it seems self.masks=masks is never reached (line 209 in the current version). Actually when batch=True, many of the plotting routines do not work properly, e.g., show_anns(...). Please check.

**Comment by giswqs** [comment, marked as answer]
You are correct. hq_sam does not support batch mode yet. Those functions were copied for the samgeo.py module. They have not been tested for batch mode. Controbutions are welcome.
